src/sentry/conf/server.py

```diff
@@ -1921,7 +1921,11 @@ def build_cdc_postgres_init_db_volume(settings):
 STATUS_PAGE_ID = None
 STATUS_PAGE_API_HOST = "statuspage.io"
 
-SENTRY_ONPREMISE = True
+# Downstream (i.e., getsentry) sets SENTRY_ONPREMISE, so keep reading from that
+# for now, but in the rest of _this_ codebase we should reference
+# SENTRY_SELF_HOSTED.
+SENTRY_ONPREMISE = True  # deprecated, use ...
+SENTRY_SELF_HOSTED = SENTRY_ONPREMISE  # ... this instead
```

Comment:
This is the risky part. We need to make sure that `getsentry` (and I guess whatever goofy third-party stuff exists out in the wild) doesn't break. As we can see, `SENTRY_ONPREMISE` defaults to true, and _probably_ nothing but `getsentry` [overrides that](https://github.com/getsentry/getsentry/blob/4a583aa9a62b78b6dcb467762a9736628fbd8711/getsentry/conf/settings/defaults.py#L29).

Reply:
Not too important for this pr, but if we want to remove `SENTRY_ONPREMISE` we should also check with the single tenant team to see whether they set this variable differently as well.

Reply:
Good call @wedamija. This came up at https://github.com/getsentry/sentry/pull/30528#discussion_r766095861 as well, and I've put single-tenant on the pile at https://github.com/getsentry/self-hosted/issues/796 to make sure it isn't forgotten. Close one!